app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -839,10 +839,17 @@ class HomeFragment : Fragment(), CoroutineScope, AccountObserver {
         }
     }
 
-    private fun SessionManager.filteredSessions(private: Boolean, sessionFilter: (Session) -> Boolean): List<Session> {
+    private fun SessionManager.filteredSessions(
+        private: Boolean,
+        sessionFilter: ((Session) -> Boolean)? = null
+    ): List<Session> {
         return this.sessions
             .filter { private == it.private }
-            .filter(sessionFilter)
+            .also { list: List<Session> ->
+                sessionFilter?.let {
```

Comment:
Won't a safe call sessionFilter?.let {} return null if the filter is null? We're promising callers a not nullable list of Sessions.

Reply:
Nevermind. It appears that the expression value isn't relevant, since also is returns `this` regardless.